ts/src/base/Exchange.ts

```diff
@@ -7423,6 +7423,51 @@ export default class Exchange {
         return optionStructures;
     }
 
+    /**
+     * @method
+     * @description checks whether the market is spot and the requested params refer to margin
+     * @param {object} params
+     * @param {string} methodName unified method name
+     * @param {Market} market
+     * @returns {boolean} true or false
+     */
+    isSpotMargin (params, methodName, market): boolean {
```

Comment:
instead of `market` should we get the `type` already? you can provide `market['type']` upon calling it.

Beisdes that we should try to type the args